app/main.py

```diff
@@ -1,25 +1,90 @@
 from abc import ABC
+from typing import Union, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Union["SlideLimitationValidator", "Visitor"],
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: Union["SlideLimitationValidator", "Visitor"],
+            owner: Union["SlideLimitationValidator", "Visitor"]
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: Union["SlideLimitationValidator", "Visitor"],
+            value: int
+    ) -> None:
+        if isinstance(value, int):
+            if self.min_amount <= value <= self.max_amount:
+                setattr(instance, self.protected_name, value)
+                return
+            raise ValueError
 
 
 class Visitor:
-   pass
+    age = IntegerRange(4, 60)
+    height = IntegerRange(80, 220)
+    weight = IntegerRange(20, 120)
+
+    def __init__(
+            self, name: str, age: int, height: int, weight: int
+    ) -> None:
+        self._name = name
+        self._age = age
+        self._height = height
+        self._weight = weight
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self, age: int, height: int, weight: int
+    ) -> None:
+        self.age = age
+        self.height = height
+        self.weight = weight
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
+
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        super().__init__(age, height, weight)
```

Comment:
This is works by default. Redundant code remove it the same inside AdultSlideLimitationValidator